frontend/src/metabase/common/hooks/use-entity-query/use-entity-query.ts

```diff
@@ -61,10 +61,11 @@ export const useEntityQuery = <TId, TItem, TQuery>(
   const error = useSelector(state => getError(state, options));
 
   const dispatch = useDispatch();
-  useEffect(() => {
+  useDeepCompareEffect(() => {
     if (entityId != null && enabled) {
       const query = { ...entityQuery, id: entityId };
-      dispatch(fetch(query, { reload, requestType }));
+      const action = dispatch(fetch(query, { reload, requestType }));
+      Promise.resolve(action).catch(() => undefined);
```

Comment:
The thing is that `dispatch` is not guaranteed to return a promise, so we call `Promise.resolve` to convert all values to promises. If was already a promise, it would return that promise.

Reply:
Also the error is already logged so we can't do anything useful here but to ignore the error

Reply:
I think it's fine if this error was ignored before as well